Add Google OAuth Login/Signup Integration

Rashi-Dwivedi1812 opened the pull request:
## ✅ Google OAuth Integration via Supabase (Login & Signup)

### 🔧 What’s Implemented
- Integrated **Google OAuth login/signup** using Supabase Auth
- Configured Supabase client to handle third-party login (Google)
- Added "Continue with Google" button in both `LoginPage` and `SignupPage`
- Retained fallback for email/password authentication

### 📦 Dependencies Used
- `@supabase/supabase-js` ( used for Supabase client)

### 🖼️ UI Enhancements
- Updated `LoginPage` UI to match the same glassmorphic design as `SignupPage`
- Placed Google button consistently with clean layout

### Updated UI
<img width="1860" height="824" alt="image" src="https://github.com/user-attachments/assets/6af8c08b-9ae0-4d15-ad99-98a23acd8170" />


---

Closes #18

IkkiOcean commented:
@Rashi-Dwivedi1812 fix merge conflict

IkkiOcean commented:
@Rashi-Dwivedi1812 
Our backend is in Node.js, and since it is not a client-only application, we do not require Supabase (Backend as a Service). We can manage this ourselves using our own native backend.

Your implementation:
1) Removed the connection with the backend API, such as /signin and /signout.
2) Added a dependency (Supabase).

A better implementation would be:
1) Continue using the signin and signout API of the backend.
2) Create a backend API for GoogleAuth.

Note:
We are using Axios in the frontend, so we should use the Axios instance (i.e., `api` in our case) for backend calls.
Example: 

```
const res = await api.post("/api/auth/signin", { email, password });
 console.log(res.data);
```

IkkiOcean commented:
@Rashi-Dwivedi1812 UI looks clean. you need to work on backend for this.

Rashi-Dwivedi1812 commented:
> @Rashi-Dwivedi1812 Our backend is in Node.js, and since it is not a client-only application, we do not require Supabase (Backend as a Service). We can manage this ourselves using our own native backend.
> 
> Your implementation:
> 
> 1. Removed the connection with the backend API, such as /signin and /signout.
> 2. Added a dependency (Supabase).
> 
> A better implementation would be:
> 
> 1. Continue using the signin and signout API of the backend.
> 2. Create a backend API for GoogleAuth.
> 
> Note: We are using Axios in the frontend, so we should use the Axios instance (i.e., `api` in our case) for backend calls. Example:
> 
> ```
> const res = await api.post("/api/auth/signin", { email, password });
>  console.log(res.data);
> ```

Okay, I'll improve the backend integration and will continue using the login & signup API of the backend.

IkkiOcean commented:
> > @Rashi-Dwivedi1812 Our backend is in Node.js, and since it is not a client-only application, we do not require Supabase (Backend as a Service). We can manage this ourselves using our own native backend.
> > Your implementation:
> > 
> > 1. Removed the connection with the backend API, such as /signin and /signout.
> > 2. Added a dependency (Supabase).
> > 
> > A better implementation would be:
> > 
> > 1. Continue using the signin and signout API of the backend.
> > 2. Create a backend API for GoogleAuth.
> > 
> > Note: We are using Axios in the frontend, so we should use the Axios instance (i.e., `api` in our case) for backend calls. Example:
> > ```
> > const res = await api.post("/api/auth/signin", { email, password });
> >  console.log(res.data);
> > ```
> 
> Okay, I'll improve the backend integration and will continue using the login & signup API of the backend.

do not use supabase anyhow.
